lib/transaction.py

```diff
@@ -303,24 +303,28 @@ def parse_redeemScript(s):
     return m, n, x_pubkeys, pubkeys, redeemScript
 
 def get_address_from_output_script(_bytes):
-    decoded = [x for x in script_GetOp(_bytes)]
-
-    # The Genesis Block, self-payments, and pay-by-IP-address payments look like:
-    # 65 BYTES:... CHECKSIG
-    match = [ opcodes.OP_PUSHDATA4, opcodes.OP_CHECKSIG ]
-    if match_decoded(decoded, match):
-        return TYPE_PUBKEY, PublicKey.from_pubkey(decoded[0][1])
-
-    # Pay-by-Bitcoin-address TxOuts look like:
-    # DUP HASH160 20 BYTES:... EQUALVERIFY CHECKSIG
-    match = [ opcodes.OP_DUP, opcodes.OP_HASH160, opcodes.OP_PUSHDATA4, opcodes.OP_EQUALVERIFY, opcodes.OP_CHECKSIG ]
-    if match_decoded(decoded, match):
-        return TYPE_ADDRESS, Address.from_P2PKH_hash(decoded[2][1])
-
-    # p2sh
-    match = [ opcodes.OP_HASH160, opcodes.OP_PUSHDATA4, opcodes.OP_EQUAL ]
-    if match_decoded(decoded, match):
-        return TYPE_ADDRESS, Address.from_P2SH_hash(decoded[1][1])
+    try:
+        decoded = [x for x in script_GetOp(_bytes)]
+
+        # The Genesis Block, self-payments, and pay-by-IP-address payments look like:
+        # 65 BYTES:... CHECKSIG
+        match = [ opcodes.OP_PUSHDATA4, opcodes.OP_CHECKSIG ]
+        if match_decoded(decoded, match):
+            return TYPE_PUBKEY, PublicKey.from_pubkey(decoded[0][1])
+
+        # Pay-by-Bitcoin-address TxOuts look like:
+        # DUP HASH160 20 BYTES:... EQUALVERIFY CHECKSIG
+        match = [ opcodes.OP_DUP, opcodes.OP_HASH160, opcodes.OP_PUSHDATA4, opcodes.OP_EQUALVERIFY, opcodes.OP_CHECKSIG ]
+        if match_decoded(decoded, match):
+            return TYPE_ADDRESS, Address.from_P2PKH_hash(decoded[2][1])
+
+        # p2sh
+        match = [ opcodes.OP_HASH160, opcodes.OP_PUSHDATA4, opcodes.OP_EQUAL ]
+        if match_decoded(decoded, match):
+            return TYPE_ADDRESS, Address.from_P2SH_hash(decoded[1][1])
+    except:
```

Comment:
Don't you ever get nervous when doing catch-all exceptions?  They can hide new bugs... if Python weren't so typeless it wouldn't normally be a problem but... What do you think? Wouldn't it be a good idea to go to the trouble of doing `(ValueError, IndexError, ...)` etc here?

I used to do this more in the past but nowadays I don't ever do this... I've found it's more pain later versus the small amount of pain now to try and figure out which exceptions may be legitimately raised...

Reply:
Fair point